### ISSUE TITLE
Synchronous creation of massive amount of workflow instances leads to deadlock

### ISSUE THREAD

**Issue by JoHeinem** [zeebe-io/zeebe#447]
* given:
  * I start the zeebe broke version 0.2.0
  * I deploy the following process: 
[extended-order-process.bpmn.zip](https://github.com/zeebe-io/zeebe/files/1298602/extended-order-process.bpmn.zip)
* when:
  * I try to start 10 million workflow instances of that process synchronously: see [here](https://pastebin.com/raw/x98TCsUD)
* then:
  * After a couple of hundred thousand instances the zeebe client does not continue to create instances without any error message on either zeebe client or broker side.
* expected:
  * The workflow instance creation works flawlessly.
  * If there is an error, I see an error message.

**Comment by Zelldon** [comment]
@menski Seems to be still an issue. We should in any case fix that. As Johannes has written, no log statements indicate what is happen. I only see that the log files are no longer updated. If I execute the test without a payload it takes longer until the broker stucks.

**Comment by Zelldon** [comment]
Today in the morning I managed it to run for a while. It creates 3.8 G data output. Yesterday the largest was 1.8 G and mostly 400 - 600 MB. At the end the broker was closed and an exception appeared in the log.

```java
io.zeebe.client.cmd.BrokerErrorException: Request exception (INVALID_MESSAGE): Cannot deserialize command: 'Could not deserialize object. Deserialization stuck at offset 170; Could not read property 'workflowKey'.; Not a long.'.
at io.zeebe.client.impl.RequestController$FailedState.doWork(RequestController.java:443)
	at io.zeebe.client.impl.RequestController$FailedState.doWork(RequestController.java:429)
	at io.zeebe.util.state.StateMachine.doWork(StateMachine.java:111)
	at io.zeebe.client.impl.RequestController.doWork(RequestController.java:162)
	at io.zeebe.client.impl.RequestManager.doWork(RequestManager.java:82)
	at io.zeebe.util.actor.ActorRunner.tryRunActor(ActorRunner.java:165)
	at io.zeebe.util.actor.ActorRunner.runActor(ActorRunner.java:145)
	at io.zeebe.util.actor.ActorRunner.doWork(ActorRunner.java:114)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:71)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by menski** [comment]
FYI the sync creation of 10 million workflow instances finished on the Zeebe server 3 without an error. It took ~30min and used ~30GB of disk space

**Comment by menski** [comment]
Interestingly the creation of 10 million workflow instances asynchronously took 1:15h on the Zeebe server 3. Not sure if this is caused by how the test is written.

**Comment by Zelldon** [comment]
I would say because we wait until all task are completed. What do you mean?

**Comment by Zelldon** [comment]
What was the size of the log?

**Comment by menski** [comment]
> I would say because we wait until all task are completed. What do you mean?

Sorry I forgot to mention that I added another test case which also just creates workflow instance without a task subscription (see https://github.com/zeebe-io/zeebe/commit/852dcc6a4faec29d374637241f9d6056116822b0). So it basically does the same as the sync test but needs more then double the time. At the beginning of the test it is way faster but then slows down extremely. The persistent data was again around 30GB.

**Comment by Zelldon** [comment]
@menski since we fixed: #555 #556 we should rerun the sync test also with 10 million workflow instances on the Server 3.

**Comment by Zelldon** [comment]
@menski  Seems that we have a new issue. The same exception which also appear in the performance test.
```
Caused by: io.zeebe.client.cmd.ClientException: No response received (timeout 5 seconds). Request was: [ topic = default-topic, partition = 1, event type = WORKFLOW_INSTANCE ]. Request receivers: [RemoteAddress{streamId=1, addr=0.0.0.0:51015}]
	at io.zeebe.client.impl.RequestController.generateTimeoutException(RequestController.java:179)
	at io.zeebe.client.impl.RequestController$ExecuteRequestState.doWork(RequestController.java:385)
	at io.zeebe.client.impl.RequestController$ExecuteRequestState.doWork(RequestController.java:350)
	at io.zeebe.util.state.StateMachine.doWork(StateMachine.java:111)
	at io.zeebe.client.impl.RequestController.doWork(RequestController.java:163)
	at io.zeebe.client.impl.RequestManager.doWork(RequestManager.java:81)
	at io.zeebe.util.actor.ActorRunner.tryRunActor(ActorRunner.java:165)
	at io.zeebe.util.actor.ActorRunner.runActor(ActorRunner.java:138)
	at io.zeebe.util.actor.ActorRunner.doWork(ActorRunner.java:114)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:71)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Zelldon** [comment]
**Short sum:**
We think that the timeout exception reflect, which are reported at the begin of this issue, that the broker simply stops to work. Since we now have client requests timeouts, the requests end in a timeout exception. The broker is then closed in the after branch of the unit test.

We will try to reproduce this next week.

**Comment by Zelldon** [comment]
Saw the following deserialization exception again:

```java
io.zeebe.client.cmd.BrokerErrorException: Request exception (INVALID_MESSAGE): Cannot deserialize command: 'Could not deserialize object. Deserialization stuck at offset 170; Could not read property 'state'.; Not a string'.


	at io.zeebe.client.cmd.BrokerErrorException.newInCurrentContext(BrokerErrorException.java:55)
	at io.zeebe.client.impl.RequestManager.waitAndResolve(RequestManager.java:152)
	at io.zeebe.client.impl.RequestManager.execute(RequestManager.java:111)
	at io.zeebe.client.impl.cmd.CommandImpl.execute(CommandImpl.java:38)
	at io.zeebe.client.impl.cmd.CommandImpl.execute(CommandImpl.java:25)
	at io.zeebe.broker.it.workflow.LargeWorkflowTest.shouldCreateBunchOfWorkflowInstances(LargeWorkflowTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: io.zeebe.client.cmd.BrokerErrorException: Request exception (INVALID_MESSAGE): Cannot deserialize command: 'Could not deserialize object. Deserialization stuck at offset 170; Could not read property 'state'.; Not a string'.

	at io.zeebe.client.impl.RequestController$FailedState.doWork(RequestController.java:463)
	at io.zeebe.client.impl.RequestController$FailedState.doWork(RequestController.java:449)
	at io.zeebe.util.state.StateMachine.doWork(StateMachine.java:111)
	at io.zeebe.client.impl.RequestController.doWork(RequestController.java:165)
	at io.zeebe.client.impl.RequestManager.doWork(RequestManager.java:83)
	at io.zeebe.util.actor.ActorRunner.tryRunActor(ActorRunner.java:190)
	at io.zeebe.util.actor.ActorRunner.runActor(ActorRunner.java:170)
	at io.zeebe.util.actor.ActorRunner.doWork(ActorRunner.java:139)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Zelldon** [comment]
```
Full thread dump OpenJDK 64-Bit Server VM (25.144-b01 mixed mode):

"actor-scheduler-executors" #22 prio=5 os_prio=0 tid=0x00007fc090767800 nid=0x2a06 in Object.wait() [0x00007fc00caf5000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000008d397bd0> (a java.lang.Object)
	at io.zeebe.util.actor.ActorSchedulerRunnable.doWorkUntilClosed(ActorSchedulerRunnable.java:135)
	- locked <0x000000008d397bd0> (a java.lang.Object)
	at io.zeebe.util.actor.ActorSchedulerRunnable.run(ActorSchedulerRunnable.java:114)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-executors-2" #21 prio=5 os_prio=0 tid=0x00007fc090765800 nid=0x2a05 waiting on condition [0x00007fc00cbf6000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-executors-1" #20 prio=5 os_prio=0 tid=0x00007fc090764000 nid=0x2a04 waiting on condition [0x00007fc00ccf7000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-default" #19 prio=5 os_prio=0 tid=0x00007fc090763000 nid=0x2a02 waiting on condition [0x00007fc00cdf8000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-default" #18 prio=5 os_prio=0 tid=0x00007fc090669800 nid=0x2a01 waiting on condition [0x00007fc07410a000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.lang.Thread.run(Thread.java:748)

"actor-scheduler-broker" #16 prio=5 os_prio=0 tid=0x00007fc03c013000 nid=0x29fb in Object.wait() [0x00007fc0768a7000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000008c6c5e58> (a java.lang.Object)
	at io.zeebe.util.actor.ActorSchedulerRunnable.doWorkUntilClosed(ActorSchedulerRunnable.java:135)
	- locked <0x000000008c6c5e58> (a java.lang.Object)
	at io.zeebe.util.actor.ActorSchedulerRunnable.run(ActorSchedulerRunnable.java:114)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-broker-2" #15 prio=5 os_prio=0 tid=0x00007fc03c011000 nid=0x29fa waiting on condition [0x00007fc0769a8000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-broker-1" #14 prio=5 os_prio=0 tid=0x00007fc03c010000 nid=0x29f9 waiting on condition [0x00007fc076aa9000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:338)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:130)
	at org.agrona.concurrent.BackoffIdleStrategy.idle(BackoffIdleStrategy.java:95)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

"actor-runner-service-container" #12 prio=5 os_prio=0 tid=0x00007fc090612000 nid=0x29f7 in Object.wait() [0x00007fc0772ab000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000008c98c520> (a io.zeebe.servicecontainer.impl.WaitingIdleStrategy)
	at io.zeebe.servicecontainer.impl.WaitingIdleStrategy.idle(WaitingIdleStrategy.java:56)
	- locked <0x000000008c98c520> (a io.zeebe.servicecontainer.impl.WaitingIdleStrategy)
	at io.zeebe.servicecontainer.impl.WaitingIdleStrategy.idle(WaitingIdleStrategy.java:34)
	at io.zeebe.util.actor.ActorRunner.doWorkUntilClose(ActorRunner.java:96)
	at io.zeebe.util.actor.ActorRunner.run(ActorRunner.java:81)
	at java.lang.Thread.run(Thread.java:748)

"Service Thread" #9 daemon prio=9 os_prio=0 tid=0x00007fc0901bd000 nid=0x29f2 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread2" #8 daemon prio=9 os_prio=0 tid=0x00007fc0901ba000 nid=0x29f1 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #7 daemon prio=9 os_prio=0 tid=0x00007fc0901b5000 nid=0x29f0 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=0 tid=0x00007fc0901b4000 nid=0x29ef waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Monitor Ctrl-Break" #5 daemon prio=5 os_prio=0 tid=0x00007fc0901b7800 nid=0x29ee runnable [0x00007fc07c33c000]
   java.lang.Thread.State: RUNNABLE
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	- locked <0x000000008c9905e0> (a java.io.InputStreamReader)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.io.BufferedReader.readLine(BufferedReader.java:324)
	- locked <0x000000008c9905e0> (a java.io.InputStreamReader)
	at java.io.BufferedReader.readLine(BufferedReader.java:389)
	at com.intellij.rt.execution.application.AppMainV2$1.run(AppMainV2.java:64)

"Signal Dispatcher" #4 daemon prio=9 os_prio=0 tid=0x00007fc090181800 nid=0x29ed waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=0 tid=0x00007fc09014e800 nid=0x29e5 in Object.wait() [0x00007fc07c8f1000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000008c94dca8> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x000000008c94dca8> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" #2 daemon prio=10 os_prio=0 tid=0x00007fc09014a000 nid=0x29e3 in Object.wait() [0x00007fc07c9f2000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000008c990ee8> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x000000008c990ee8> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"main" #1 prio=5 os_prio=0 tid=0x00007fc09000d800 nid=0x29d4 waiting on condition [0x00007fc098bd8000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000f238cb20> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
	at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at io.zeebe.client.impl.RequestManager.waitAndResolve(RequestManager.java:141)
	at io.zeebe.client.impl.RequestManager.execute(RequestManager.java:111)
	at io.zeebe.client.impl.cmd.CommandImpl.execute(CommandImpl.java:38)
	at io.zeebe.client.impl.cmd.CommandImpl.execute(CommandImpl.java:25)
	at io.zeebe.broker.it.workflow.LargeWorkflowTest.shouldCreateBunchOfWorkflowInstances(LargeWorkflowTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

"VM Thread" os_prio=0 tid=0x00007fc090142800 nid=0x29e1 runnable 

"GC task thread#0 (ParallelGC)" os_prio=0 tid=0x00007fc090023000 nid=0x29d8 runnable 

"GC task thread#1 (ParallelGC)" os_prio=0 tid=0x00007fc090025000 nid=0x29da runnable 

"GC task thread#2 (ParallelGC)" os_prio=0 tid=0x00007fc090026800 nid=0x29db runnable 

"GC task thread#3 (ParallelGC)" os_prio=0 tid=0x00007fc090028800 nid=0x29dd runnable 

"VM Periodic Task Thread" os_prio=0 tid=0x00007fc0901bf000 nid=0x29f3 waiting on condition 

JNI global references: 760

Heap
 PSYoungGen      total 615936K, used 450352K [0x00000000d9580000, 0x00000000fff80000, 0x0000000100000000)
  eden space 599040K, 75% used [0x00000000d9580000,0x00000000f4d24340,0x00000000fde80000)
  from space 16896K, 0% used [0x00000000fde80000,0x00000000fdea8000,0x00000000fef00000)
  to   space 16384K, 0% used [0x00000000fef80000,0x00000000fef80000,0x00000000fff80000)
 ParOldGen       total 57856K, used 29115K [0x000000008c000000, 0x000000008f880000, 0x00000000d9580000)
  object space 57856K, 50% used [0x000000008c000000,0x000000008dc6ec58,0x000000008f880000)
 Metaspace       used 28651K, capacity 29604K, committed 29696K, reserved 1075200K
  class space    used 3375K, capacity 3567K, committed 3584K, reserved 1048576K

```

**Comment by Zelldon** [comment]
Run the sync LargeWorkflowTest repeatedly and every time when the brocker stucks the last event which was written is the Task CREATED event. Maybe that helps to determine what is wrong here.

**Comment by menski** [comment]
this should be fixed with #596
